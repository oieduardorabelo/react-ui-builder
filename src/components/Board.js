import React from 'react'
import {DropTarget} from 'react-dnd'

const dropTarget = {
  drop(props, monitor, component) {
    const section = monitor.getItem().name
    props.onDrop(section)
    return {
      name: 'Board'
    }
  }
}

@DropTarget('Section', dropTarget, (connect, monitor) => ({
  connectDropTarget: connect.dropTarget(),
  isOver: monitor.isOver(),
  canDrop: monitor.canDrop()
}))
class Board extends React.Component {
  static propTypes = {
    connectDropTarget: React.PropTypes.func.isRequired,
    isOver: React.PropTypes.bool.isRequired,
    canDrop: React.PropTypes.bool.isRequired,
    children: React.PropTypes.array.isRequired
  }

  render() {
    const { connectDropTarget } = this.props
    return (connectDropTarget(
      <div className='Board'>{this.props.children}</div>
    ))
  }
}

export default Board
