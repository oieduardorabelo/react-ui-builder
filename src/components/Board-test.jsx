import React from 'react'

import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Board from './Board'

describe('<Board />', () => {
  it('renders correctly', () => {
    const OriginalBoard = Board.DecoratedComponent
    const identity = el => el
    const wrapper = shallow(
      <OriginalBoard connectDropTarget={identity} drop>
        {[]}
      </OriginalBoard>,
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders with one child correctly', () => {
    const OriginalBoard = Board.DecoratedComponent
    const identity = el => el
    const wrapper = shallow(
      <OriginalBoard connectDropTarget={identity} drop>
        {[<span key="1">Fake</span>]}
      </OriginalBoard>,
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders with multiple child correctly', () => {
    const OriginalBoard = Board.DecoratedComponent
    const identity = el => el
    const wrapper = shallow(
      <OriginalBoard connectDropTarget={identity} drop>
        {[
          <span key="1">Fake</span>,
          <span key="2">Fake</span>,
          <span key="3">Fake</span>,
        ]}
      </OriginalBoard>,
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
