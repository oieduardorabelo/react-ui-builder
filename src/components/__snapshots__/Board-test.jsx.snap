// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board /> renders correctly 1`] = `
<div
  className="Board"
/>
`;

exports[`<Board /> renders with multiple child correctly 1`] = `
<div
  className="Board"
>
  <span>
    Fake
  </span>
  <span>
    Fake
  </span>
  <span>
    Fake
  </span>
</div>
`;

exports[`<Board /> renders with one child correctly 1`] = `
<div
  className="Board"
>
  <span>
    Fake
  </span>
</div>
`;
